lib/index.js

```diff
@@ -1,16 +1,83 @@
-function updateWeather(cityState) {
+const initialVisit = () => {
+  if(document.cookie) {
+    let cookies = document.cookie.split('; ');
+    apiKey = cookies[0].split('=')[1];
+    let username = cookies[1].split('=')[1];
+    if(apiKey != '') {
+      let usernameElementParent = document.getElementById('hello-user');
+      let usernameElement = document.createElement('p');
+      let usernameText = document.createTextNode('Welcome, ' + username);
+      usernameElement.appendChild(usernameText);
+      usernameElementParent.appendChild(usernameElement);
```

Comment:
JS is really tricky for things like this, it works great for a small project, but these areas become a huge PITA once you start transitioning into maintenance mode. One thing you can do to combat this is to start separating the logic out into “display” logic and “event” logic. Display logic hides and shows specific things, while the event logic handles when to do it.

Reply:
Sounds good. Hopefully I can refactor a ton of stuff once I finish basic functionality. It definitely has too much going on and it's just a tiny page. It's already difficult to find what I want, although it is organized from top down more or less which helps. Looking forward to working in a framework again.